modules/core/keeper/msg_server_test.go

```diff
@@ -868,9 +868,9 @@ func (suite *KeeperTestSuite) TestChannelUpgradeTry() {
 				suite.Require().NotNil(res)
 				suite.Require().Equal(channeltypes.FAILURE, res.Result)
 
-				// TODO: assert error receipt exists for the upgrade sequence when RestoreChannel / AbortUpgrade is called
-				// errorReceipt, found := suite.chainB.GetSimApp().GetIBCKeeper().ChannelKeeper.GetUpgradeErrorReceipt(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
-				// suite.Require().True(found)
+				errorReceipt, found := suite.chainB.GetSimApp().GetIBCKeeper().ChannelKeeper.GetUpgradeErrorReceipt(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+				suite.Require().True(found)
+				suite.Require().Equal(uint64(1), errorReceipt.Sequence)
```

Comment:
> but incompatible proposed upgrades does

where is this currently happening?

My understanding of this expected value resulting in `uint64(1)` is that `channel.UpgradeSequence` is 0 before TRY is invoked, and because we are moving from OPEN -> TRY it calls `chanUpgradeInit` [here](https://github.com/cosmos/ibc-go/blob/04-channel-upgrades/modules/core/04-channel/keeper/upgrade.go#L124) which increments the sequence 

> if upgrade sequence is always bumped on init, I don't see a reason to bump our own upgrade sequence

Agree, but I find it hard to imagine the fast forward scenario, having a hard time understanding what could cause these sequences to be completely out of sync.

Reply:
I see, yup that's correct!